app/main.py

```diff
@@ -1,3 +1,62 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    @staticmethod
+    def type_check(elem: Distance | int | float) -> int | float | None:
+        if not isinstance(elem, Distance | int | float):
+            raise TypeError("Other object should be Distance, "
+                            "int or float type!")
+        if isinstance(elem, Distance):
+            elem = elem.km
+        return elem
```

Comment:
unecessary if here - there are only 2 variants - it will be the instance of Distance or not 
Also change name of variable to more informative - not elem f.e element, value or smth like that